pkg/apis/planetscale/v2/vitessbackupstorage_types.go

```diff
@@ -116,6 +118,27 @@ type S3BackupLocation struct {
 	AuthSecret *SecretSource `json:"authSecret,omitempty"`
 }
 
+// AzblobBackupLocation specifies a backup location in Amazon S3
+type AzblobBackupLocation struct {
+	// Account is the name of the Azure storage account to use
+	// +kubebuilder:validation:MinLength=1
+	Account string `json:"account"`
+	// Container is the name of the Azure storage account container to use.
+	// +kubebuilder:validation:MinLength=1
+	Container string `json:"container"`
+	// KeyPrefix is an optional prefix added to all object keys created by Vitess.
+	// This is only needed if the same container is also used for something other
+	// than backups for VitessClusters. Backups from different clusters,
+	// keyspaces, or shards will automatically avoid colliding with each other
+	// within a container, regardless of this setting.
+	// +kubebuilder:validation:Pattern=^[^\r\n]*$
+	// +kubebuilder:validation:MaxLength=256
+	KeyPrefix string `json:"keyPrefix,omitempty"`
+	// AuthSecret is a reference to the Secret to use for the Azure authentication
+	// The secret must contain two keys; azblob-account-key, and azblob-account-name
+	AuthSecret *SecretSource `json:"authSecret"`
```

Comment:
In AWS and GCP, we make the AuthSecret optional (`omitempty`) for those who choose to give implicit blob store permissions to the Nodes themselves. Does the Vitess Azure plugin support such a mode, or is this truly required?

If it's truly required, let's make this a non-pointer type. If we decide to make it optional instead, we'll have to add `nil` checks down in `storage_azblob.go`.

Reply:
It is in fact required